framework/wazuh/core/cluster/hap_helper/proxy.py

```diff
@@ -651,6 +670,34 @@ async def exists_frontend(self, frontend_name: str) -> bool:
         """
         return frontend_name in await self.get_current_frontends()
 
+    async def check_multiple_frontends(self, port: int) -> bool:
+        """Check if there are multiple frontends binding the given port.
+
+        Parameters
+        ----------
+        port : int
+            Port number to check.
+
+        Returns
+        -------
+        bool
+            True if exists mutiple frontends else False.
+        """
+        self.logger.debug(f'Checking multiple frontends for port {port}')
+        frontends = await self.get_current_frontends()
+        port_bind_exists = False
+
+        for frontend in frontends.keys():
+            data = (await self.api.get_binds(frontend=frontend))['data']
+            binds = [bind for bind in data if bind.get('port') == port]
+
+            if binds and port_bind_exists:
+                return True
+            elif binds:
+                port_bind_exists = True
+
+        return False
```

Comment:
This would return False if there is only one bind to 1514, but the helper will later add its own 1514 bind so there will be two. This should return True always that one bind with the same port is found (as long as the frontend name is different to `f'{backend_name}_front'`)

Reply:
I modified the behavior of the check. 

Now, when we start the proxy for the first time having a frontend binding the same port. We got the warning

![image](https://github.com/wazuh/wazuh/assets/7127104/607274ff-cb0d-49e0-a954-6d0615746cf7)

```console
root@wazuh-master:/var/ossec# framework/python/bin/python3 framework/scripts/wazuh_clusterd.py -f
2024/05/23 17:28:56 DEBUG: [Cluster] [Main] Removing '/var/ossec/queue/cluster/'.
2024/05/23 17:28:56 DEBUG: [Cluster] [Main] Removed '/var/ossec/queue/cluster/'.
Starting cluster in foreground (pid: 7360)
2024/05/23 17:28:57 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/05/23 17:28:57 DEBUG: [Local Server] [Keep alive] Calculating.
2024/05/23 17:28:57 DEBUG: [Local Server] [Keep alive] Calculated.
2024/05/23 17:28:57 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
2024/05/23 17:28:57 DEBUG: [Master] [Keep alive] Calculating.
2024/05/23 17:28:57 DEBUG: [Master] [Keep alive] Calculated.
2024/05/23 17:28:57 INFO: [Master] [Local integrity] Starting.
2024/05/23 17:28:57 INFO: [Master] [Local agent-groups] Sleeping 30s before starting the agent-groups task, waiting for the workers connection.
2024/05/23 17:28:57 INFO: [HAPHelper] [Main] Proxy was initialized
2024/05/23 17:28:57 DEBUG: [HAPHelper] [Proxy] Checking multiple frontends for port 1514
2024/05/23 17:28:57 DEBUG2: [HAPHelper] [Proxy] Obtained proxy frontends
2024/05/23 17:28:57 WARNING: [HAPHelper] [Main] Several frontends exist binding the port "1514". To ensure the proper function of the helper, keep only the one related to the backend "wazuh_reporting".
2024/05/23 17:28:57 DEBUG2: [HAPHelper] [Proxy] Obtained proxy backends
2024/05/23 17:28:57 INFO: [HAPHelper] [Main] Could not find Wazuh backend 'wazuh_reporting'
2024/05/23 17:28:57 INFO: [Master] [Local integrity] Finished in 0.089s. Calculated metadata of 34 files.
2024/05/23 17:28:57 DEBUG2: [HAPHelper] [Proxy] Added new proxy backend: 'wazuh_reporting'
2024/05/23 17:28:57 INFO: [HAPHelper] [Main] Added Wazuh backend
2024/05/23 17:28:57 DEBUG2: [HAPHelper] [Proxy] Obtained proxy frontends
```

After that, the `wazuh_reporting_front` will be created

![image](https://github.com/wazuh/wazuh/assets/7127104/ba20accf-b1fe-4580-a002-b5f58b3111f9)

But, when we start the helper and `wazuh_reporting_front` already exists, the warning will not show

![image](https://github.com/wazuh/wazuh/assets/7127104/436189cf-87f2-40ea-9617-14f14ab87fff)


```console
root@wazuh-master:/var/ossec# framework/python/bin/python3 framework/scripts/wazuh_clusterd.py -f
2024/05/23 17:32:09 DEBUG: [Cluster] [Main] Removing '/var/ossec/queue/cluster/'.
2024/05/23 17:32:09 DEBUG: [Cluster] [Main] Removed '/var/ossec/queue/cluster/'.
Starting cluster in foreground (pid: 9606)
2024/05/23 17:32:10 INFO: [Local Server] [Main] Serving on /var/ossec/queue/cluster/c-internal.sock
2024/05/23 17:32:10 DEBUG: [Local Server] [Keep alive] Calculating.
2024/05/23 17:32:10 DEBUG: [Local Server] [Keep alive] Calculated.
2024/05/23 17:32:10 INFO: [Master] [Main] Serving on ('0.0.0.0', 1516)
2024/05/23 17:32:10 DEBUG: [Master] [Keep alive] Calculating.
2024/05/23 17:32:10 DEBUG: [Master] [Keep alive] Calculated.
2024/05/23 17:32:10 INFO: [Master] [Local integrity] Starting.
2024/05/23 17:32:10 INFO: [Master] [Local agent-groups] Sleeping 30s before starting the agent-groups task, waiting for the workers connection.
2024/05/23 17:32:10 INFO: [HAPHelper] [Main] Proxy was initialized
2024/05/23 17:32:10 DEBUG: [HAPHelper] [Proxy] Checking multiple frontends for port 1514
2024/05/23 17:32:10 DEBUG2: [HAPHelper] [Proxy] Obtained proxy frontends
2024/05/23 17:32:10 INFO: [HAPHelper] [Main] Ensuring only exists one HAProxy process. Sleeping 30s before start...
2024/05/23 17:32:10 INFO: [Master] [Local integrity] Finished in 0.090s. Calculated metadata of 34 files.
2024/05/23 17:32:12 INFO: [Worker] [Main] Connection from ('172.31.0.3', 49134)
2024/05/23 17:32:12 DEBUG: [Worker] [Main] Command received: b'hello'
```